gtsam/nonlinear/Values-inl.h

```diff
@@ -197,6 +197,59 @@ namespace gtsam {
      }
    };
 
+#ifdef _WIN32
+   // Handle dynamic matrices
+   template <int M, int N>
+   struct handle_matrix<Eigen::Matrix<double, M, N, 0, M, N>, true> {
```

Comment:
I don't really understand why this is not in the right case, it may indicate a bigger problem which is the default case being different in different compilation units...

Reply:
I don't understand it either. My suspicion that is a msvc compiler bug.
On gcc compiler on this code, I get ambiguate as I expected.

Reply:
What does the full MSVC substitution look like?

Reply:
I don't know. I just guess this one according to the input.

Reply:
Can you please add a comment here about what this is doing and why is it needed? The comment can state that this is due to a possible MSVC bug.

Reply:
I added